### PR TITLE
kernel: dlist: add test cases to cover all APIs

### PR DESCRIPTION
**Pull request by Zhaoningx**
Add a testcase to test the APIs which not
covered by existing testcases.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

### CODE REVIEW

File: tests/unit/list/dlist.c [hunk 1]
```diff
@@ -327,6 +327,90 @@ void test_dlist(void)
...
+	memset(test_node, 0, sizeof(test_node));
+	memset(&insert_node, 0, sizeof(insert_node));
+	memset(&insert_node2, 0, sizeof(insert_node2));
+	sys_dlist_init(&test_list);
+
+	/* Check if the dlist is empty */
+	zassert_true(sys_dlist_get(&test_list) == NULL,
+			"Get a empty dilst, NULL will be returned");
+
+	/* Check if insert node inta a empty dlist */
```
Comment (LeiW000) [inline tests/unit/list/dlist.c:358]: It's better to change it like below

"Check if a node can be inserted into an empty dlist."

File: tests/unit/list/dlist.c [hunk 2]
```diff
@@ -327,6 +327,90 @@ void test_dlist(void)
...
+
+	zassert_true(sys_dlist_peek_prev(&test_list,
+				NULL) == NULL,
+				"dlist appended incorrectly");
+
+	zassert_true(sys_dlist_get(&test_list) ==
+				&test_node[0].node,
+				"Get a dilst, head will be returned");
+
+	/* Check after inserting into a known node */
```
Comment (LeiW000) [inline tests/unit/list/dlist.c:401]: I guess what you meant is that inserting a known node into a dlist. Same to line 407.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/unit/list/dlist.c (hunk 2 above)
Comment (Zhaoningx) [reply tests/unit/list/dlist.c]: Hi @LeiW000 , this three parts you mentioned all test the same API sys_dlist_insert_at(), the API's means, insert node at a position you want. My comments were not too clear, so I changed them again. Do you think the current comments make sense?

**Review by LeiW000 — APPROVED** [review]

**Comment by LeiW000** [comment]
@enjiamai, please review the PR.

**Review by enjiamai — APPROVED** [review]

**Comment by LeiW000** [comment]
@nashif , could you help to merge it?

**Pull request merged**